pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2191,10 +2198,19 @@ func (a *AWSCloud) getInstancesByNodeNames(nodeNames []string) ([]*ec2.Instance,
 // Returns the instance with the specified node name
 // Returns nil if it does not exist
 func (a *AWSCloud) findInstanceByNodeName(nodeName string) (*ec2.Instance, error) {
+	var filterKey string
+	if a.cfg.Global.UseKubernetesNodeTag {
+		filterKey = "tag:" + TagNameKubernetesNode
+	} else {
+		filterKey = "private-dns-name"
+	}
+
 	filters := []*ec2.Filter{
-		newEc2Filter("private-dns-name", nodeName),
+		newEc2Filter(filterKey, nodeName),
+		newEc2Filter("instance-state-name", "running"),
```

Comment:
It's more of an issue with the node labels, since there will be duplicates and it will fail.

Without node labels, just searching on 'KubernetesCluster', it will potentially attach terminated instances. I don't think this is an issue as health check will fail and it won't route traffic.

I am happy to make a separate PR though and can do that, since it is somewhat an orthogonal change.